### DISCUSSION TITLE
CmdHelp `commands` and `rules` arguments

### DISCUSSION THREAD

**Discussion by davewiththenicehat** [evennia/evennia#2444] [Evennia Engine Development]
On my list of updates to the help system is allowing CmdHelp to accept arguments `commands` and `rules`. As with the other updates I am making to the help system, I wanted to offer commiting the changes back to evennia. Since this is one of the few places where my changes to evennia may be something universally helpful.

If CmdHelp receives the `commands` argument it will show only the Command help entries.
If CmdHelp receives the `rules` argument it will only show database and file help entries.

Because this would override all help topics, I would make variables in `evennia.settings_default` to toggle these options.
By override I mean `help commands` would only show all the command help entries. It would not show a file, database or Command help entry named `commands`.

**Comment by Griatch** [comment]
You can already view help entries by category. Why is another such split needed? If you want to have your 'rules' easily viewable, just put your rules as 'rules' category and do `help rules` ?

**Reply by davewiththenicehat** [reply]
It is part of what I am doing to make the help command more approachable by first time text gamers.

It allows developers to replace the output of CmdHelp with no arguments. Than still have easy access to all the entries without the player needing to know categories.
It also allows a method for showing commands separate from other entries. Where categories will show both.

Just formatting suggestions I am happy to commit.
The message on CmdHelp is a big one. Wanted to offer what I am doing as an option to make it easy to segrigate.

**Comment by Griatch** [comment]
Thanks for the suggestions, but no I don't think there is a need to be able to segregate commands from other help entries by default. They could all be mixed together if so desired too, and a game dev can tweak the `CmdHelp` class to customize this if they so desire.
